src/web/traceViewer/ui/snapshotTab.tsx

```diff
@@ -27,16 +27,18 @@ export const SnapshotTab: React.FunctionComponent<{
   snapshotSize: Size,
 }> = ({ action, snapshotSize }) => {
   const [measure, ref] = useMeasure<HTMLDivElement>();
-  let [snapshotIndex, setSnapshotIndex] = React.useState(0);
+  const [snapshotIndex, setSnapshotIndex] = React.useState(0);
 
   const snapshotMap = new Map<string, { title: string, snapshotName: string }>();
   for (const snapshot of action?.metadata.snapshots || [])
     snapshotMap.set(snapshot.title, snapshot);
   const actionSnapshot = snapshotMap.get('action') || snapshotMap.get('after');
   const snapshots = [actionSnapshot ? { ...actionSnapshot, title: 'action' } : undefined, snapshotMap.get('before'), snapshotMap.get('after')].filter(Boolean) as { title: string, snapshotName: string }[];
 
-  if (snapshotIndex >= snapshots.length)
-    snapshotIndex = snapshots.length - 1;
+  React.useEffect(() => {
```

Comment:
we could also move this into the parameter of useState so its used as a default value. So we could get rid of one re-render.